presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -0,0 +1,545 @@
+                        metadata.getFunctionRegistry(),
+                        new PreGeneratedExpressions(tryMethodMap.build(), lambdaFieldMap.build()));
+
+                List<Parameter> inputParameters = ImmutableList.<Parameter>builder()
+                        .add(session)
+                        .add(page)
+                        .add(position)
+                        .build();
+
+                MethodDefinition tryMethod = defineTryMethod(
+                        innerExpressionVisitor,
+                        containerClassDefinition,
+                        "try_" + counter,
+                        inputParameters,
+                        Primitives.wrap(tryExpression.getType().getJavaType()),
+                        tryExpression,
+                        callSiteBinder);
+
+                tryMethodMap.put(tryExpression, tryMethod);
+            }
+            else if (lambdaOrTryExpression instanceof LambdaDefinitionExpression) {
+                LambdaDefinitionExpression lambdaExpression = (LambdaDefinitionExpression) lambdaOrTryExpression;
+                PreGeneratedExpressions preGeneratedExpressions = new PreGeneratedExpressions(tryMethodMap.build(), lambdaFieldMap.build());
+                FieldDefinition methodHandleField = LambdaBytecodeGenerator.preGenerateLambdaExpression(
+                        lambdaExpression,
+                        "lambda_" + counter,
+                        containerClassDefinition,
+                        preGeneratedExpressions,
+                        callSiteBinder,
+                        cachedInstanceBinder,
+                        metadata.getFunctionRegistry());
+                lambdaFieldMap.put(lambdaExpression, methodHandleField);
+            }
+            else {
+                throw new VerifyException(format("unexpected expression: %s", lambdaOrTryExpression.toString()));
+            }
+            counter++;
+        }
+
+        return new PreGeneratedExpressions(tryMethodMap.build(), lambdaFieldMap.build());
+    }
+
+    private static void generateConstructor(
+            ClassDefinition classDefinition,
+            CachedInstanceBinder cachedInstanceBinder,
+            Consumer<MethodDefinition> additionalStatements)
+    {
+        MethodDefinition constructorDefinition = classDefinition.declareConstructor(a(PUBLIC));
+
+        BytecodeBlock body = constructorDefinition.getBody();
+        Variable thisVariable = constructorDefinition.getThis();
+
+        body.comment("super();")
+                .append(thisVariable)
+                .invokeConstructor(Object.class);
+
+        additionalStatements.accept(constructorDefinition);
+
+        cachedInstanceBinder.generateInitializations(thisVariable, body);
+        body.ret();
+    }
+
+    private static RowExpressionVisitor<Scope, BytecodeNode> fieldReferenceCompiler(CallSiteBinder callSiteBinder)
+    {
+        return new InputReferenceCompiler(
```

Comment:
Do you have some benchmark results for this change? Previously channels were accessed through dedicated:
`block_` fields. Won't accessing rows through `page.getBlock(x).getPosition(y)` be a performance bottleneck for tight computations now?